Hippo.Web/ClientApp/src/components/Account/ApproveAccounts.test.tsx

```diff
@@ -0,0 +1,206 @@
+  if ((global.fetch as any).mockClear) {
+    (global.fetch as any).mockClear();
+  }
+  unmountComponentAtNode(container);
+  container.remove();
+});
+
+it("shows pending approvals count", async () => {
+  await act(async () => {
+    render(
+      <AppContext.Provider value={(global as any).Hippo}>
+        <ModalProvider>
+          <MemoryRouter>
+            <ApproveAccounts />
+          </MemoryRouter>
+        </ModalProvider>
+      </AppContext.Provider>,
+      container
+    );
+  });
+  expect(container.textContent).toContain(
+    "There are 2 account(s) awaiting your approval"
+  );
+});
+
+it("shows approval button for each pending account", async () => {
+  await act(async () => {
+    render(
+      <AppContext.Provider value={(global as any).Hippo}>
+        <ModalProvider>
+          <MemoryRouter>
+            <ApproveAccounts />
+          </MemoryRouter>
+        </ModalProvider>
+      </AppContext.Provider>,
+      container
+    );
+  });
+  expect(container.querySelectorAll("button.btn.btn-primary").length).toBe(2);
+});
+
+it("shows reject button for each pending account", async () => {
+  await act(async () => {
+    render(
+      <AppContext.Provider value={(global as any).Hippo}>
+        <ModalProvider>
+          <MemoryRouter>
+            <ApproveAccounts />
+          </MemoryRouter>
+        </ModalProvider>
+      </AppContext.Provider>,
+      container
+    );
+  });
+  expect(container.querySelectorAll("button.btn.btn-danger").length).toBe(2);
+});
+
+it("approve button has expected text", async () => {
+  await act(async () => {
+    render(
+      <AppContext.Provider value={(global as any).Hippo}>
+        <ModalProvider>
+          <MemoryRouter>
+            <ApproveAccounts />
+          </MemoryRouter>
+        </ModalProvider>
+      </AppContext.Provider>,
+      container
+    );
+  });
+  expect(container.querySelector("button.btn.btn-primary")?.textContent).toBe(
+    "Approve"
+  );
+});
+it("reject button has expected text", async () => {
+  await act(async () => {
+    render(
+      <AppContext.Provider value={(global as any).Hippo}>
+        <ModalProvider>
+          <MemoryRouter>
+            <ApproveAccounts />
+          </MemoryRouter>
+        </ModalProvider>
+      </AppContext.Provider>,
+      container
+    );
+  });
+  expect(container.querySelector("button.btn.btn-danger")?.textContent).toBe(
+    "Reject"
+  );
+});
+
+it("table header has expected text", async () => {
+  await act(async () => {
+    render(
+      <AppContext.Provider value={(global as any).Hippo}>
+        <ModalProvider>
+          <MemoryRouter>
+            <ApproveAccounts />
+          </MemoryRouter>
+        </ModalProvider>
+      </AppContext.Provider>,
+      container
+    );
+  });
+  expect(container.querySelector("tr")?.textContent).toBe(
+    "NameSubmittedAction"
+  );
+});
+
+//Enable this when the test works
+xit("displays dialog when reject is clicked", async () => {

```

Comment:
Not sure how to verify the reject dialog shows up in the test. skipping this test for now.